spacy/cli/init_config.py

```diff
@@ -1,55 +1,64 @@
 from ..util import SimpleFrozenList
-from ._util import init_cli, Arg, Opt, show_validation_error, COMMAND
-from ._util import string_to_list, import_code, _handle_renamed_language_codes
+from ._util import cli, convert_string_list, show_validation_error, COMMAND
+from ._util import import_code, _handle_renamed_language_codes
 
 
 ROOT = Path(__file__).parent / "templates"
 TEMPLATE_PATH = ROOT / "quickstart_training.jinja"
 RECOMMENDATIONS = srsly.read_yaml(ROOT / "quickstart_training_recommendations.yml")
 
 
-class Optimizations(str, Enum):
-    efficiency = "efficiency"
-    accuracy = "accuracy"
+OptimizationsType = Literal["efficiency", "accuracy"]
 
 
 class InitValues:
     """
-    Default values for initialization. Dedicated class to allow synchronized default values for init_config_cli() and
-    init_config(), i.e. initialization calls via CLI respectively Python.
+    Default values for initialization. Dedicated class to allow synchronized
+    default values for init_config_cli() and init_config(), i.e. initialization
+    calls via CLI respectively Python.
     """
 
-    lang = "en"
-    pipeline = SimpleFrozenList(["tagger", "parser", "ner"])
-    optimize = Optimizations.efficiency
-    gpu = False
-    pretraining = False
-    force_overwrite = False
+    lang: str = "en"
+    pipeline: List[str] = SimpleFrozenList(["tagger", "parser", "ner"])
+    optimize: OptimizationsType = "efficiency"
+    gpu: bool = False
+    pretraining: bool = False
+    force_overwrite: bool = False
 
 
-@init_cli.command("config")
-def init_config_cli(
+@cli.subcommand(
+    "init",
+    "config",
     # fmt: off
-    output_file: Path = Arg(..., help="File to save the config to or - for stdout (will only output config and no additional logging info)", allow_dash=True),
-    lang: str = Opt(InitValues.lang, "--lang", "-l", help="Code of the language to use"),
-    pipeline: str = Opt(",".join(InitValues.pipeline), "--pipeline", "-p", help="Comma-separated names of trainable pipeline components to include (without 'tok2vec' or 'transformer')"),
-    optimize: Optimizations = Opt(InitValues.optimize, "--optimize", "-o", help="Whether to optimize for efficiency (faster inference, smaller model, lower memory consumption) or higher accuracy (potentially larger and slower model). This will impact the choice of architecture, pretrained weights and related hyperparameters."),
-    gpu: bool = Opt(InitValues.gpu, "--gpu", "-G", help="Whether the model can run on GPU. This will impact the choice of architecture, pretrained weights and related hyperparameters."),
-    pretraining: bool = Opt(InitValues.pretraining, "--pretraining", "-pt", help="Include config for pretraining (with 'spacy pretrain')"),
-    force_overwrite: bool = Opt(InitValues.force_overwrite, "--force", "-F", help="Force overwriting the output file"),
+    output_file=Arg(help="File to save the config to or - for stdout (will only output config and no additional logging info)"),
```

Comment:
Wouldn't it be nicer to have a convertor function here that returns a `Path` also for the `"-"` option, so the rest of the code can be agnostic to it? In `radicli` there's the function `convert_path_or_dash` but it returns a `Union[Path, str]` instead of converting to Path in both cases, why is that?

Reply:
It would make the typing more consistent but it kinda felt weird because `Path("-")` is not really a valid path, it's kinda just a `Literal["-"]` symbol to indicate stdin/stdout.

(I also briefly considered having a converter that already opens the file, but this can get problematic, because we often want different handling depending on the file type vs. stdin/stdout. Like, if it's a JSON file, load as JSON, if it's JSONL, load as JSONL, if it's stdin/stdout, treat it as JSONL etc.)